src/main/java/mekanism/client/SparkleAnimation.java

```diff
@@ -16,59 +18,69 @@
-            World world = tile.getWorld();
-            ThreadLocalRandom random = ThreadLocalRandom.current();
+        World world = tile.getWorld();
+        ThreadLocalRandom random = ThreadLocalRandom.current();
+        int sparkles = 0;
 
-            for (MutableBlockPos pos : itr) {
-                TileEntity t = world.getTileEntity(pos);
-                if (!nodeChecker.isNode(t)) {
-                    continue;
-                }
+        while (itr.hasNext()) {
+            BlockPos pos = itr.next();
+            if (world.isAirBlock(pos)) {
+                continue;
+            }
 
-                for (int i = 0; i < 6; i++) {
-                    world.spawnParticle(EnumParticleTypes.REDSTONE, pos.getX() + random.nextDouble(),
-                          pos.getY() + -.01, pos.getZ() + random.nextDouble(), 0, 0, 0);
-                    world.spawnParticle(EnumParticleTypes.REDSTONE, pos.getX() + random.nextDouble(),
-                          pos.getY() + 1.01, pos.getZ() + random.nextDouble(), 0, 0, 0);
-                    world.spawnParticle(EnumParticleTypes.REDSTONE, pos.getX() + random.nextDouble(),
-                          pos.getY() + random.nextDouble(), pos.getZ() + -.01, 0, 0, 0);
-                    world.spawnParticle(EnumParticleTypes.REDSTONE, pos.getX() + random.nextDouble(),
-                          pos.getY() + random.nextDouble(), pos.getZ() + 1.01, 0, 0, 0);
-                    world.spawnParticle(EnumParticleTypes.REDSTONE, pos.getX() + -.01,
-                          pos.getY() + random.nextDouble(), pos.getZ() + random.nextDouble(), 0, 0, 0);
-                    world.spawnParticle(EnumParticleTypes.REDSTONE, pos.getX() + 1.01,
-                          pos.getY() + random.nextDouble(), pos.getZ() + random.nextDouble(), 0, 0, 0);
-                }
+            TileEntity t = world.getTileEntity(pos);
+            if (t == null || !nodeChecker.isNode(t)) {
+                continue;
             }
-        }).start();
+
+            for (int i = 0; i < 2; i++) {
+                sparkles++;
+                world.spawnParticle(EnumParticleTypes.REDSTONE, pos.getX() + random.nextDouble(),
+                      pos.getY() + random.nextDouble(), pos.getZ() + -.01, 0, 0, 0);
+                world.spawnParticle(EnumParticleTypes.REDSTONE, pos.getX() + random.nextDouble(),
+                      pos.getY() + random.nextDouble(), pos.getZ() + 1.01, 0, 0, 0);
+                world.spawnParticle(EnumParticleTypes.REDSTONE, pos.getX() + -.01,
+                      pos.getY() + random.nextDouble(), pos.getZ() + random.nextDouble(), 0, 0, 0);
+                world.spawnParticle(EnumParticleTypes.REDSTONE, pos.getX() + 1.01,
+                      pos.getY() + random.nextDouble(), pos.getZ() + random.nextDouble(), 0, 0, 0);
+            }
+        }
+
+        long elapsed = System.nanoTime() - start;
+        Mekanism.logger.info("Generated {} sparkles in {} ms", sparkles, elapsed / 1000000d);
```

Comment:
Is this still some debug info?

Reply:
Yes - I left that in there while the review is ongoing. Will remove before final commit.